compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -291,26 +345,25 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             | ty::InstanceDef::CloneShim(..)
             | ty::InstanceDef::Item(_) => {
                 // We need MIR for this fn
-                let body =
+                let (body, instance) =
```

Comment:
Ah, this change to record the actual instance of the callee in the stack seems to break some const-eval panic tests... I guess I can make the CTFE machine's `find_mir_or_eval_fn` return the original instance here, but having the MIR Body and the instance not match does seem rather sketchy in terms of the FnAbi checks.